### PR TITLE
Smart contract interactions

### PR DESCRIPTION
**Pull request by jumperdevweb3**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -109,13 +110,35 @@ model Erc20Token {
...
+
+model SmartContractInteraction {
+  id                    String                       @id @default(cuid())
+  DevelopersUserAccount DevelopersUserAccount        @relation(fields: [developerUserId], references: [id])
+  developerUserId       String
+  SmartContract         SmartContract                @relation(fields: [smartContractId], references: [id])
+  smartContractId       String
+  method                String
+  txHash                String
+  output                Json?
```
Comment (adiusz) [inline prisma/schema.prisma:133]: wer input :D

File: src/app/api/public/contracts/[contractName]/[usersContractVersion]/[functionName]/route.ts [hunk 2]
```diff
@@ -106,6 +107,24 @@ async function postHandler(req: NextRequestWithAuth, { params: { contractName, u
 
       const transactionResult = await gaslessTransaction(account, calldata);
 
+      const txRes = await provider.waitForTransaction(transactionResult.transactionHash) as any;
+      console.log("Interaction response: ",txRes)
+      const fee = parseInt((txRes as any)?.actual_fee?.amount, 16);
```
Comment (adiusz) [inline src/app/api/public/contracts/[contractName]/[usersContractVersion]/[functionName]/route.ts:112]: is it correctly formatting fees to wei? asking becasue i have no idea and want to double-check with you

File: src/app/api/public/contracts/[contractName]/route.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {NextRequest, NextResponse} from "next/server";
+import {getContractsFunctions} from "@/contracts/interfaces";
+import {StatusCodes} from "http-status-codes";
+import {withDeveloperApiToken} from "@/app/middleware/withDeveloperApiToken";
+
+async function handler(req: NextRequest, { params: { contractName } }){
+   const functions = getContractsFunctions(contractName);
+
+    return NextResponse.json({functions},{status: StatusCodes.OK});
```
Comment (adiusz) [inline src/app/api/public/contracts/[contractName]/route.ts:9]: extra space 😇

File: src/app/api/public/contracts/route.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { NextResponse } from "next/server";
-import { withDevUserApiToken } from "@/app/middleware/withDevUserApiToken";
 import { StatusCodes } from "http-status-codes";
 import { getAllContractsDetails } from "@/contracts/interfaces";
 import { smartContractModel } from "@/prisma/models";
+import {withDeveloperApiToken} from "@/app/middleware/withDeveloperApiToken";
```
Comment (adiusz) [inline src/app/api/public/contracts/route.ts:5]: unformatted import

**Review by adiusz — CHANGES_REQUESTED** [review]

File: src/app/api/public/contracts/[contractName]/[usersContractVersion]/[functionName]/route.ts (hunk 2 above)
Comment (jumperdevweb3) [reply src/app/api/public/contracts/[contractName]/[usersContractVersion]/[functionName]/route.ts:112]: Yes

**Pull request merged**